### PR TITLE
add readme for code samples

### PR DESCRIPTION
**Pull request by furkansenharputlu**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 * [Installing the Client](#installing-the-client)
 * [Using the Client](#using-the-client)
 * [Serialization Considerations](#serialization-considerations)
+* [Code Samples](#code-samples)
```
Comment (mustafaiman) [inline README.md]: Can we move Code Samples above `Serialization Considerations`? I think `Code Samples` is higher priority than it.

File: code_samples/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+**global_serializer.js** — creating and adding global serializer to serialization config
+
+**lifecycle_listener.js** — listening to lifecycle events of the Hazelcast instance
+
+**list.js** — some basic list operations
+
+**logging.js** — defining log levels
+
+**map.js** — some basic map operations
```
Comment (mustafaiman) [inline code_samples/README.md]: I think `some basic` is not neccessary. We could go with `usage of distributed [map/flakeid/list] etc` where appropriate.

File: code_samples/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Code Samples
+
+This document explains Hazelcast Node.js Client code samples which you can use to learn how to use Hazelcast features.
```
Comment (Serdaro) [inline code_samples/README.md]: This folder (`/code_samples`) contains an extensive collection of Hazelcast Node.js Client code samples, which helps you to learn how to use Hazelcast features. The following lists the samples with their brief descriptions.

File: code_samples/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+**org-website/** — samples to display on https://hazelcast.org/
```
Comment (Serdaro) [inline code_samples/README.md]: ... - Folder including the samples to display on https://hazelcast.org.

File: code_samples/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+**aggregation.js** — some built-in aggregation samples
```
Comment (Serdaro) [inline code_samples/README.md]: ..... - Built-in aggregation samples.

File: code_samples/README.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+**custom_serializer.js** — creating and adding custom serializer to serialization config
```
Comment (Serdaro) [inline code_samples/README.md]: ..... - Creating ...... serialization configuration.

File: code_samples/README.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+**distibuted_object_listener.js** — listening to distributed object events
```
Comment (Serdaro) [inline code_samples/README.md]: ..... - Listening ...... events.

File: code_samples/README.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+**flakeid_generator.js** — usage of distributed flakeid generator
```
Comment (Serdaro) [inline code_samples/README.md]: ...... - Usage ..... Flake ID Generator.

File: code_samples/README.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+**global_serializer.js** — creating and adding global serializer to serialization config
```
Comment (Serdaro) [inline code_samples/README.md]: ..... - Creating ..... serialization configuration.

File: code_samples/README.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+**lifecycle_listener.js** — listening to lifecycle events of the Hazelcast instance
```
Comment (Serdaro) [inline code_samples/README.md]: ..... - Listening ..... instance.

File: code_samples/README.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+**list.js** — usage of distributed list
```
Comment (Serdaro) [inline code_samples/README.md]: ..... - Usage ..... list.

File: code_samples/README.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+**logging.js** — defining log levels
```
Comment (Serdaro) [inline code_samples/README.md]: .... - Defining log levels.

File: code_samples/README.md [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+**map.js** — usage of distributed map
```
Comment (Serdaro) [inline code_samples/README.md]: ..... - Usage ..... map.

File: code_samples/README.md [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+
+**map_entry_listener.js** — listening to map events
```
Comment (Serdaro) [inline code_samples/README.md]: ..... - Listening to map events.

File: code_samples/README.md [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+
+**near_cache.js** — usage of distributed nearcached map
```
Comment (Serdaro) [inline code_samples/README.md]: .... - Usage of distributed map with a near cache.

File: code_samples/README.md [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+
+**paging_predicate.js** — getting a page-by-page result of query
```
Comment (Serdaro) [inline code_samples/README.md]: ..... - Getting ..... query.

File: code_samples/README.md [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+
+**pn_counter.js** — usage of distributed pn counter
```
Comment (Serdaro) [inline code_samples/README.md]: ..... - Usage of distributed PN Counter.

File: code_samples/README.md [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+
+**queue.js** — usage of distributed queue
```
Comment (Serdaro) [inline code_samples/README.md]: ...... - Usage of distributed queue.

File: code_samples/README.md [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+
+**set.js** — usage of distributed set
```
Comment (Serdaro) [inline code_samples/README.md]: ..... - Usage of distributed set.

File: code_samples/README.md [hunk 20]
```diff
@@ -0,0 +1,37 @@
...
+
+**ssl_authentication.js** — mutual authentication to cluster via ssl
```
Comment (Serdaro) [inline code_samples/README.md]: ..... - Mutual .....via SSL.

**Pull request merged**